src/v3/src/OktaSignIn/index.ts

```diff
@@ -236,10 +233,34 @@ export default class OktaSignIn {
   getUser(): void { }
 
   on(eventName: OktaWidgetEventType, eventHandler: OktaWidgetEventHandler): void {
-    this.events[eventName] = eventHandler;
+    let registeredEventHandler = eventHandler;
+    if (EVENTS_LIST.includes(eventName)) {
```

Comment:
should we trap third party errors for all events and not just ones that are included in `EVENTS_LIST`?

Reply:
We can. But other events are not public. I tried to keep parity with Gen2:
https://github.com/okta/okta-signin-widget/blob/master/src/widget/OktaSignIn.ts#L192

For handlers of public events (there are only 3 of those: `ready`, `afterRender`, `afterError`) we need to catch errors to prevent  throwing unhandled errors in SIW components (because event callbacks are executed synchronously in React lifecycle method `useEffect`).

There is currently one private event (undocumented, developers should not use, used in SIW code) - `hide`: https://github.com/okta/okta-signin-widget/blob/master/src/v3/src/components/Widget/index.tsx#L402
There are also private events in Gen1/2 (like `route`) that developers could theoretically subscribe on, and if their handlers throw error, it would be uncaught.

Reply:
I see okay yeah that does make sense that with events open to the public this is definitely needed.  Where as with private events we are/should be in control.  I think if we are in parity with gen2 then it should be okay.

@shawyu-okta any opinion on this?